util/src/main/scala/org/ensime/util/path.scala

```diff
@@ -0,0 +1,29 @@
+package org.ensime.util
+
+import java.net.{ JarURLConnection, URL }
+import java.nio.charset.Charset
+import java.nio.file.{ Path => JPath }
+
+import com.google.common.io.Files
+/**
+ * Created by harsh on 15/06/16.
+ */
+package object path {
+  type Path = JPath
+  implicit class EnrichPath(val path: Path) extends AnyVal {
+    def isJarUrl = path.toString.startsWith("jar:file:")
+    def isJava: Boolean = path.toString.endsWith(".java")
+    def isScala: Boolean = path.toString.endsWith(".scala")
+    def readString(implicit cs: Charset): String = {
+      if (!path.isJarUrl) Files.toString(path.toFile, cs) else {
+        val in = new URL(path.toString).openConnection().asInstanceOf[JarURLConnection].getInputStream
```

Comment:
I'm surprised that we'd need to do it this way. I thought the `Path` API was able to handle any content. We might need to be careful about closing the jar after we've used it though.

Try to avoid using `scala.io.Source`, it's incredibly badly designed: both from a performance perspective and from a resource management / error handling perspective.